### PR TITLE
ENG-593: Improve linting in Github Action

### PR DESCRIPTION
**Pull request by triwats**
## Description

* Fixing YAML Syntax
* Implement linting via tflint

## Issue(s)

[ENG-593](https://www.notion.so/oaknationalacademy/Improve-HCL-linting-d3ed3d5b5c4c463eb41147cdae93502f?pvs=4)

### CODE REVIEW

File: actions/terraform-checks/action.yml [hunk 1]
```diff
@@ -10,6 +9,29 @@ runs:
       with:
         terraform_version: ${{ inputs.terraform_version }}
 
+    - name: Cache plugin dir
+      uses: actions/cache@v4
+      with:
+        path: ~/.tflint.d/plugins
+        key: tflint-${{ hashFiles('.tflint.hcl') }}
+
+    - name: "INFO: Setup TFLint"
```
Comment (simonrose121) [inline actions/terraform-checks/action.yml]: Is `INFO:` a yamllint thing?

Comment (triwats) [reply actions/terraform-checks/action.yml]: It's more a way to differentiate between logs, maybe not necessary, hangover from wanting to switch between commands with find (CMD+F) for example in a raw file.

Comment (simonrose121) [reply actions/terraform-checks/action.yml]: What other options would we have? I'm thinking if every step is `INFO:` then is it necessary?

File: actions/terraform-checks/action.yml [hunk 2]
```diff
@@ -20,6 +42,12 @@ runs:
         cd infrastructure
         terraform fmt -check $(find . -name '*tf')
 
+    - name: "INFO: Checking Terraform Format of Files"
+      shell: bash
+      run: |
+        cd infrastructure
+        terraform fmt -check $(find . -name '*tf')
```
Comment (tweakster) [inline actions/terraform-checks/action.yml]: Am I missing something? Isn't this the step from above repeated?

File: actions/terraform-checks/action.yml [hunk 3]
```diff
@@ -10,6 +9,29 @@ runs:
...
+      shell: bash
+      run: tflint --version
+
+    - name: "INFO:  Init TFLint"
+      shell: bash
+      run: tflint --init
+
+    - name: "INFO:  Run TFLint"
+      shell: bash
+      run: tflint -f compact
```
Comment (tweakster) [inline actions/terraform-checks/action.yml]: Do we know this checks all files. If I run this in `Cloud-Config` root dir I get nothing, but if I run this in `Cloud-Config/infrastructure/cloudflare/misc` I get
```
3 issue(s) found:

dns_records.tf:938:5: Warning - Interpolation-only expressions are deprecated in Terraform v0.12.14 (terraform_deprecated_interpolation)
:0:0: Warning - terraform "required_version" attribute is required (terraform_required_version)
access_groups.tf:148:3: Warning - local.access_groups_not_defined is declared but not used (terraform_unused_declarations)
```

File: actions/terraform-checks/action.yml (hunk 1 above)
Comment (triwats) [reply actions/terraform-checks/action.yml]: Good point, just looked at the way this is presented here: https://github.com/oaknational/oak-terraform-actions/actions/runs/10352739526/job/28654004307

image

Seems fairly unnecessary given that format.

I'll remove them 👍

Comment (triwats) [reply actions/terraform-checks/action.yml]: Removed: 
image

File: actions/terraform-checks/action.yml (hunk 2 above)
Comment (triwats) [reply actions/terraform-checks/action.yml]: Not sure how I managed this - removed repeat and kept the better name!

File: actions/terraform-checks/action.yml (hunk 3 above)
Comment (triwats) [reply actions/terraform-checks/action.yml]: Think I was testing this in the wrong directory... 

I've changed this to:

```
tflint --chdir infrastructure/ --recursive
```

Which gives me 91 errors in Cloud Config 😭 - will add another task in to fix those

Comment (triwats) [reply actions/terraform-checks/action.yml]: https://www.notion.so/oaknationalacademy/Fix-Cloud-Config-TFLint-Errors-9a6f09f6f8e44e52b6c24859bf543f97

**Review by tweakster — APPROVED** [review]

**Pull request merged**